crates/matrix-sdk/src/encryption/mod.rs

```diff
@@ -890,6 +894,40 @@ impl Encryption {
         task.await.expect("Task join error")
     }
 
+    /// Open the [`SecretStore`] with the given `key`.
+    ///
+    /// The `secret_storage_key` can be a passphrase or a Base58 encoded secret
+    /// storage key.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use matrix_sdk::Client;
+    /// # use url::Url;
+    /// # async {
+    /// # let homeserver = Url::parse("http://example.com")?;
+    /// # let client = Client::new(homeserver).await?;
+    /// use ruma::events::secret::request::SecretName;
+    ///
+    /// let secret_store = client
+    ///     .encryption()
+    ///     .open_secret_store("It's a secret to everybody")
+    ///     .await?;
+    ///
+    /// let my_secret = "Top secret secret";
+    /// let my_secret_name = SecretName::from("m.treasure");
+    ///
+    /// secret_store.store_secret(&my_secret_name, my_secret);
+    ///
+    /// # anyhow::Ok(()) };
+    /// ```
+    pub async fn open_secret_store(
+        &self,
+        secret_storage_key: &str,
+    ) -> secret_storage::Result<SecretStore> {
+        SecretStore::open(self.client.to_owned(), secret_storage_key).await
```

Comment:
Is it fine to create multiple `SecreStore`? Shouldn't `Client` own an instance of `SecretStore` and `open_secret_store` to return a reference to it instead?

Reply:
Should be fine to open multiple ones. The client can't really own one, at least not a open one since it needs user provided key to open the store.